backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -532,30 +613,89 @@ public FutureExperimentRunDAO(
               return localQueryContext;
             },
             executor);
-
-    final var futurePredicatesContext =
-        predicatesHandler.processPredicates(request.getPredicatesList(), executor);
-
-    final var futureSortingContext =
-        sortingHandler.processSort(request.getSortKey(), request.getAscending());
-
-    final var futureProjectIds =
-        getAllowedProjects(ModelDBActionEnum.ModelDBServiceActions.READ)
-            .thenApply(
-                projIds ->
-                    new QueryFilterContext()
-                        .addCondition("experiment_run.project_id in (<authz_project_ids>)")
-                        .addBind(q -> q.bindList("authz_project_ids", projIds)),
-                executor);
+    queryFilterContextList.add(futureLocalContext);
+
+    // futurePredicatesContext
+    queryFilterContextList.add(
+        predicatesHandler.processPredicates(request.getPredicatesList(), executor));
+
+    // futureSortingContext
+    queryFilterContextList.add(
+        sortingHandler.processSort(request.getSortKey(), request.getAscending()));
+
+    // futureProjectIds based on workspace
+    if (request.getWorkspaceName().isEmpty()) {
+      queryFilterContextList.add(
+          getAllowedProjects(ModelDBActionEnum.ModelDBServiceActions.READ)
+              .thenCompose(
+                  projectIds -> {
+                    if (projectIds.isEmpty()) {
+                      // TODO: This is not works, need to update it.
+                      /*return InternalFuture.failedStage(new PermissionDeniedException(
+                      "Accessible projects not found in workspace: "
+                              + request.getWorkspaceName()));*/
```

Comment:
Why doesn't it work?

Reply:
Fixed in PR https://github.com/VertaAI/modeldb/pull/2242